### ISSUE TITLE
Calculated/Adjusted Values Persist in Milestone Report Form Data After Changing Milestone Type to Reporting

### ISSUE THREAD

**Issue by gurjmatharu** [bcgov/cas-cif#1745]
Describe the Bug:
When you have a milestone report, with calculated and/or adjusted data.  Then you change the milestone type to reporting, the formdata still has the calculated values.  They show up on the diff screen.

https://www.loom.com/share/8a62e10419a14496b6a7dcd67ffdf8fa

Probability (how likely the bug is to happen, scored from 1-5): 5

Effect (how bad the bug is when it does happen, scored from 1-5): 5

**Comment by Sepehr-Sobhani** [comment]
When using the `update_milestone_form_change` mutation, we currently set calculated values. To improve this, we should modify it so that it only passes fields that are relevant to the milestone type and exclude the others.

**Comment by pbastia** [comment]
> One fix strategy would be the user shouldn't be able to switch milestone types with payments to one without payments during the editing process.

@gurjmatharu this would be fixing a bug at the expense of the user, not fixing the code. Could we just remove that extra data when we switch milestone type?

**Comment by gurjmatharu** [comment]
> > One fix strategy would be the user shouldn't be able to switch milestone types with payments to one without payments during the editing process.
> 
> @gurjmatharu this would be fixing a bug at the expense of the user, not fixing the code. Could we just remove that extra data when we switch milestone type?

Yup, that works.  I'm just thinking that it's odd that we would allow the user to enter this state.  If we want to support a milestone report where payment info is filled out, and then later on, switch milestone types to a non payment, then for sure.  The question I have is this an expected part of the user journey?  Should there be an extra alert, saying "you will lose payment info if you do this?"  We have similar messaging elsewhere in the app, or is this not a big deal and we can just remove the extra data?
